src/components/TopNav.js

```diff
@@ -36,9 +41,27 @@ class TopNav extends Component {
     this.setState({ redirectToLogin: true });
   }
 
+  handleMenuOpen = (event) => {
+    this.setState({
+      userMenuAnchorElement: event.target,
+    });
+  };
+
+  handleMenuClose = () => {
+    this.setState({
+      userMenuAnchorElement: null,
+    });
+  };
+
   render() {
     const { userName } = this.props;
-    const { redirectToLogin } = this.state;
+    const { redirectToLogin, userMenuAnchorElement } = this.state;
+
+    const NavBarSpacer = withStyles((theme) => ({
```

Comment:
Adding this margin directly to the `AppBar` somehow broke the user drop-down menu. I think this is probably about as good.